channeldb/revocation_log.go

```diff
@@ -103,15 +154,20 @@ func (h *HTLCEntry) toTlvStream() (*tlv.Stream, error) {
 	)
 
 	return tlv.NewStream(
-		tlv.MakePrimitiveRecord(rHashType, &h.RHash),
+		tlv.MakeDynamicRecord(
+			rHashType, &h.RHash, h.RHashLen,
+			RHashEncoder, RHashDecoder,
+		),
 		tlv.MakePrimitiveRecord(
 			refundTimeoutType, &h.RefundTimeout,
 		),
 		tlv.MakePrimitiveRecord(
 			outputIndexType, &h.OutputIndex,
 		),
 		tlv.MakePrimitiveRecord(incomingType, &h.incomingTlv),
-		tlv.MakePrimitiveRecord(amtType, &h.amtTlv),
+		// We will save 3 bytes if the amount is less or equal to
```

Comment:
Ah ok I see it here, np to have this as another commit since we lay out the initial structure in the first commit.

Reply:
What about the refund timeout as well?

Reply:
Thought about it too. Since we are already over the block height of 65,535, it would require 5 bytes to encode it using `bigsize`. In this case it saves more bytes to use `uint32`.